llvm/include/llvm/Transforms/Vectorize/SandboxVectorizer/SandboxVectorizer.h

```diff
@@ -17,10 +21,22 @@ class TargetTransformInfo;
 class SandboxVectorizerPass : public PassInfoMixin<SandboxVectorizerPass> {
   TargetTransformInfo *TTI = nullptr;
 
-public:
-  PreservedAnalyses run(Function &F, FunctionAnalysisManager &AM);
+  // Used to build a RegionPass pipeline to be run on Regions created by the
+  // bottom-up vectorization pass.
+  sandboxir::PassRegistry PR;
+
+  // The main vectorizer pass.
+  std::unique_ptr<sandboxir::BottomUpVec> BottomUpVecPass;
```

Comment:
Why is this a member variable? It's created and used only within `runOnFunction()` at the moment, so making it a class member implies that it used across multiple member functions of `SandboxVectorizerPass`. Are you planning to move it to a separate class that `SandboxVectorizerPass` inherits from?

Reply:
if we move away from `cl::opt("sbvec-passes")` and to proper pass parameters, we would want the LLVM pass to hold the region pass manager in the class, rather than keep it local to `runImpl`, because we'd specify the region pass pipeline via the constructor

but with `cl::opt` I don't think it really matters

Reply:
actually, I thought that `BottomUpVecPass` would own the region pass manager so it could invoke it when it creates a region

Reply:
> making it a class member implies that it used across multiple member functions of SandboxVectorizerPass

Not the only reason possible to make something a member. Making it local to `runImpl` means constructing the `BottomUpVec` pass every time, which seems like a waste to me.

Also, conceptually I see the `SandboxVectorizer` pass as a wrapper around `BottomUpVec` that handles the creation of sandbox IR. I don't see anything wrong with the wrapper owning the thing it's wrapping.

Reply:
> Not the only reason possible to make something a member. Making it local to runImpl means constructing the BottomUpVec pass every time, which seems like a waste to me.

Ah yes, I forgot that we were building the pass pipeline over and over again.

Reply:
> actually, I thought that BottomUpVecPass would own the region pass manager so it could invoke it when it creates a region

I missed `BottomUpVec` also having a reference to the region pass manager